bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseAudioAudioSource.java

```diff
@@ -95,14 +93,9 @@ public void close() throws IOException {
                     };
                     registerPipe(pipeOutput);
                     // get raw audio from the pulse audio socket
-                    return new PulseAudioStream(sourceFormat, pipeInput, (idle) -> {
-                        setIdle(idle);
-                        if (idle) {
-                            scheduleDisconnect();
-                        } else {
-                            // ensure pipe is writing
-                            startPipeWrite();
-                        }
+                    return new PulseAudioStream(sourceFormat, pipeInput, () -> {
```

Comment:
I do not understand this change. Why is the consumed value ignored in the new implementation?

Reply:
After reviewing the previous code I think it should be:
```
if (!idle) {
     // ensure pipe is writing
     startPipeWrite();
}
```

Reply:
I change the Consumer<Boolean> to a Runnable : 
The Consumer<Boolean> is, as I understand it, a sort of on/off switch that the PulseAudioStream can call to activate or stop the  process of copying data from the source.
In the new method I try to use, I want to centralize the place where we can cause a (smart) disconnection. So, I want the PulseAudioStream to only decrement the counter.
And it seems that the "off" switch part of the Consumer is not necessary, because it was called from the PulseAudioStream.close() method, and this method calls the close() method of the PipedInputStream, which in turn can act as the central method to decrement the counter properly by calling unregisterPipe().

And by removing the need for an "off" switch, remains only the need for a "on" switch, i.e. an activity function. Which is why I change the Consumer<Boolean> to a Runnable called "activity".
I think it should be possible to also delete this "activity" runnable, because starting the startPipeWrite method once should be sufficient, but I thought "better safe than sorry".

You propose : 
`if (!idle) {
     // ensure pipe is writing
     startPipeWrite();
}`
I would like to remove the global isIdle boolean as it introduced side effect I explained in the open post. A count of active client should be safer.
And the activity Runnable is now only called from the read() method of the PulseAudioStream : so when we call it, we never are in an idle state. So the test seems not needed. Or there is something I didn't think of, which is totally possible ?

Thank for your implication with my bug.

Reply:
Thanks for the explanation, I was missing the consumer change to Runnable, now I see it clear.